lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1925,8 +1925,12 @@ void IRGenDebugInfoImpl::emitVariableDeclaration(
 
     // There are variables without storage, such as "struct { func foo() {}
     // }". Emit them as constant 0.
-    if (isa<llvm::UndefValue>(Piece))
-      Piece = llvm::ConstantInt::get(IGM.Int64Ty, 0);
+    if (isa<llvm::UndefValue>(Piece)) {
```

Comment:
What was this change about?

Reply:
The existing behavior is broken for an `undef` of any type that isn't 64 bits in size. I should break it out into its own commit.